### DISCUSSION TITLE
[Lua] load_plus 加载文件增强

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#106] [INFO 信息]
# 本文已过时

参见 https://github.com/hooke007/MPV_lazy/commit/ccda13df4808622a471941cf07727a12a145b412

<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>

## 说明

**[load_plus.lua](https://github.com/hooke007/MPV_lazy/blob/main/portable_config/scripts/load_plus.lua)** 是基于旧版的 [autoload.lua](https://github.com/mpv-player/mpv/blob/master/TOOLS/lua/autoload.lua) & [open_dialog.lua](https://github.com/rossy/mpv-open-file-dialog) 合并改进而来。

普通脚本，使用常规的安装方式即可。  [【通用引导参考】](https://hooke007.github.io/unofficial/mpv_start.html#id14)
默认不内建预绑定的快捷键，按照例子即可快速设置 input.conf
https://github.com/hooke007/MPV_lazy/blob/8cdbf554f7ceac3eaa7fe7e1057d13c5f882487e/portable_config/scripts/load_plus.lua#L14-L18

## 功能

查看其对应的示范脚本设置文件 **[load_plus.conf](https://github.com/hooke007/MPV_lazy/blob/main/portable_config/script-opts/load_plus.conf)** 了解细分设定

- 列表中仅存在单文件时自动补充同路径下的其它文件到条目中。
  - 可以使用自定义的扩展名白名单
- 调用原生windows的对话框加载文件/地址、追加音轨、字幕。

### 缺陷

- 不支持取消加载URL的对话框，否则会连带中止mpv的进程，如果误触，即使地址留空也应点击确认。

**Comment by 921j** [comment]
mpv-lazy-20220721 下不起作用
今天从仓库下了最新版还是不行
换成原来 autoload.lua 立即生效

**Reply by hooke007** [reply]
哪个功能不起作用
如果你是指和autoload重合的功能，不存在不生效的问题，懒人版的预设和脚本的默认值不同
https://github.com/hooke007/MPV_lazy/blob/f9ef63cd32da9852779c2b3b0bb0d05304b8196d/portable_config/script-opts/load_plus.conf#L4-L5

**Reply by 921j** [reply]
> 哪个功能不起作用
抱歉，刚发完就发现是自己没看懂配置而且默认的是加载近似文件名，把
load_plus.conf 里的
level=1 改为 level=-1 就跟 autoload.lua 是一样了
建议默认还是-1得好，我测试的目录里没有文件名近似的文件，搞得我以为没起作用

**Reply by 921j** [reply]
> ##自动填充的等级 —— 按用户的条件（默认 -1）/ 始终阻止，0 / 仅近似名文件，1 

这个说明我断成了  （默认 -1）/ 始终阻止
                                 0 / 仅近似名文件

映像里值 -1 一般都是“禁止”的意思，所以半天没想到是这个的问题

**Reply by hooke007** [reply]
默认本来就是 -1
要使用默认设置删掉conf就行。预设值是我的使用习惯为主，不会修改

**Comment by eko5624** [comment]
Mac端load_plus.lua能增加从url打开的功能吗。我看官方mac端有这样的工具栏菜单：File>Open URL...

**Reply by hooke007** [reply]
我对mac的接口和语法一窍不通

**Reply by dyphire** [reply]
load_plus.lua 中集成的 open_dialog.lua 相关代码实现全是 win 端专有，非 win 端不推荐使用此脚本，直接使用原版 autoload.lua 即可。

加载文件/其它音轨/其它字幕的功能可由 uosc.lua 或 [mpv-file-browser](https://github.com/CogentRedTester/mpv-file-browser) 平替，加载 url 的功能可通过 [SmartCopyPaste_II.lua](https://github.com/Eisa01/mpv-scripts/blob/master/scripts/SmartCopyPaste_II.lua) 代替

至于次字幕滤镜功能可以通过写 uosc.lua 脚本插件实现

**Comment by CNhjx** [comment]
level=1 时脚本不起作用，是设置的问题吗
![image](https://user-images.githubusercontent.com/57735696/213351169-6bfa42ed-eb3d-44c1-880d-abef5027898d.png)

**Reply by hooke007** [reply]
不是，这个的应用场景是文件夹内混杂了不同作品的系列剧集。纯数字不在考虑范围内

**Reply by CNhjx** [reply]
文件命名是有什么要求吗，有些可以有些不行
![image](https://user-images.githubusercontent.com/57735696/213406236-86ad18ae-ecfd-47d2-9138-5931d1891f9f.png)
![image](https://user-images.githubusercontent.com/57735696/213406487-9da0039d-d215-4776-9e3f-04abbd2372e5.png)

**Reply by hooke007** [reply]
语义上有歧义，这个功能的确适用性是比较狭窄的，毕竟不是常见的功能。

这是某个动画.01
这是某个动画.02
这是某个动画.03
这是另一部电视剧.01
这是另一部电视剧.02
这是另一部电视剧.03

**Comment by hooke007** [comment]
此帖终结，改用 https://mpv.io/manual/master/#options-autocreate-playlist
